.buildkite/pipeline.yml

```diff
@@ -42,6 +42,7 @@ steps:
           .buildkite/commands/install-node-dependencies.sh
           node ./scripts/prepare-dev-build-version.mjs
           npm run make:macos-{{matrix}}
+          npm run make:dmg-{{matrix}}
```

Comment:
Should it happen after we notarize the binary?

Reply:
Here's what [the docs](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow#3087734:%7E:text=Alternatively%2C%20you%20can%20put,the%20items%20it%20downloads.) say: 

> Alternatively, you can put apps, kernel extensions, and other software in a container, like a disk image, and notarize the container. The notary service accepts disk images (UDIF format), signed flat installer packages, and ZIP archives. It processes nested containers as well, like packages inside a disk image.
>
> **Important:** If you distribute your software via a custom third-party installer, you need two rounds of notarization. First you notarize the installer’s payload (everything the installer will install). You then package the notarized (and stapled, as described in Staple the ticket to your distribution) items into the installer and notarize it as you would any other executable. If you use a network installer, separately notarize both the installer and the items it downloads.

So, I think in our case, given we distribute both `.app` and `.dmg`, we'll want to notarize both. If we were only distributing the DMG, we could notarize just that and trust Apple's tooling will handle it.

Reply:
I tried a local notarization of a DMG built from this branch, as of d538a1714bd489bbcc842880834d1249ddc624cd, and this diff worked:

```diff
--- a/fastlane/Fastfile
+++ b/fastlane/Fastfile
@@ -45,13 +45,20 @@ lane :set_up_signing do |_options|
 end

 desc 'Notarize the compiled binary'
-lane :notarize_binary do |_options|
+lane :notarize_binary do
   Dir[File.join(BUILDS_FOLDER, '**', 'Studio.app')].each do |path|
     notarize(
       package: path,
       api_key_path: APPLE_API_KEY_PATH
     )
   end
+  Dir[File.join(BUILDS_FOLDER, '**', 'Studio-*.dmg')].each do |path|
+    notarize(
+      bundle_id: APPLE_BUNDLE_IDENTIFIER,
+      package: path,
+      api_key_path: APPLE_API_KEY_PATH
+    )
+  end
 end

 desc 'Ship the binary to internal testers'
```